### ISSUE TITLE
No lint failure when fragment variable is not defined

### ISSUE THREAD

**Issue by beaucollins** [dotansimha/graphql-eslint#652]
**Describe the bug**
- Given two `.graphql` files, one with a `query` operation and another with a `fragment`
- And `fragment` uses a variable of a required input
- And `query` does not define input variable

Then: there is no lint failure

(inverse of #651 )

**To Reproduce**

```
git clone -b fragment-with-variable git@github.com:beaucollins/
npm install
npm run lint
```

**Expected behavior**

The same error as if the fragment was embedded within the query:

```
./graphql-lint-fragment-variables/src/GetWidgets.graphql
  9:36  error  Variable "$usageInterval" is not defined by operation "AllWidgets"  @graphql-eslint/no-undefined-variables
```

**Environment:**

- OS: macOS 11.5.2
- `@graphql-eslint/eslint-plugin@2.2.0`:
- NodeJS: v16.9.1

**Comment by B2o5T** [comment]
I don't get errors from eslint, get you verify your branch?

**Comment by beaucollins** [comment]
> I don't get errors from eslint, can you verify your branch?

Correct, not having an eslint error here is the problem.

It doesn't produce a lint error but it should. The fragment requires a variable but the linter doesn't understand that when the fragment is defined elsewhere.

As an example copy the contents of the fragment in place within the `query` operation and it will show an error.

**Comment by beaucollins** [comment]
Applying this diff will show the error that is expected. The contents of the frargment in the other file when copied over into the same file as the operation shows the expected failure.

```diff
diff --git a/src/GetWidgets.graphql b/src/GetWidgets.graphql
index 6ff3f78..d20d660 100644
--- a/src/GetWidgets.graphql
+++ b/src/GetWidgets.graphql
@@ -1,7 +1,16 @@
-#import WidgetWithTags from './WidgetWithTags.graphql'
 query AllWidgets {
   widgets {
     id
     ...WidgetWithTags
   }
 }
+
+fragment WidgetWithTags on Widget {
+  id
+  name
+  tags {
+    id
+    name
+    count: usageCount(interval: $usageInterval)
+  }
+}
```

```
graphql-lint-fragment-variables/src/GetWidgets.graphql
  14:34  error  Variable "$usageInterval" is not defined by operation "AllWidgets"  @graphql-eslint/no-undefined-variables
```

**Comment by B2o5T** [comment]
Hi @beaucollins, thank you so much for contributing :) I fixed the false-negative case and I added a test, this will be also fixed in the next release
